js/components/discoveryPanel.js

```diff
@@ -28,9 +26,39 @@ var DiscoveryPanel = React.createClass({
         })
       }),
       icons: React.PropTypes.objectOf(React.PropTypes.object)
+    }),
+    controller: React.PropTypes.shape({
+      sendDiscoveryClickEvent: React.PropTypes.func
     })
   },
 
+  getDefaultProps: function () {
```

Comment:
make the component less fragile by ensuring props have values that may not have been specified by parent component.

better summary [here](https://facebook.github.io/react/docs/reusable-components.html#default-prop-values)